Prog/AddNew/TOPA_match_notices_property.sas

```diff
@@ -0,0 +1,98 @@
+/**************************************************************************
+ Program:  TOPA_match_notices_property.sas
+ Library:  PresCat
+ Project:  Urban-Greater DC
+ Author:   Elizabeth Burton
+ Created:  12/13/2022
+ Version:  SAS 9.4
+ Environment:  Local Windows session (desktop)
+ 
+ Description:  Match TOPA notices to the same properties
+ 
+ Modifications:
+**************************************************************************/
+
+%include "\\sas1\DCdata\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( PresCat )
+%DCData_lib( DHCD )
+%DCData_lib( MAR )
+%DCData_lib( RealProp )
+
+%File_info( data=PresCat.TOPA_SSL, printobs=5 )
+%File_info( data=PresCat.TOPA_realprop, printobs=5 ) /** some IDs don't have real_prop info**/
+%File_info( data=PresCat.TOPA_addresses, printobs=5 )
+%File_info( data=PresCat.TOPA_database, printobs=5 ) /** 1699 obs**/
+
+/** merge address_id ssl and real_prop info **/
+
+data Topa_address_ssl_realprop;
+  merge
+    PresCat.TOPA_SSL (keep=id address_id ssl in=in_ssl)
+    PresCat.TOPA_addresses (keep =id address casd_date offer_sale_date in=in_addresses);
+  by id; 
+  /** List in LOG notices that are not in both data sets **/
+  if not ( in_ssl and in_addresses ) then put in_ssl= in_addresses= id= ssl= address_id= address=;
+run;
+
+%File_info( data=Topa_address_ssl_realprop, printobs=5 )
+
+/** sorting by id then address id**/
+proc sort data=Topa_address_ssl_realprop out=Topa_address_ssl_realprop_sort;
+  by id address_id;
+run;
+
+%File_info( data=Topa_address_ssl_realprop_sort, printobs=5 )
+
+/** create ID (notice) x address_id crosswalk **/
+data Topa_id_x_address_1; 
+  set Topa_address_ssl_realprop_sort;
+  by id; 
+  if first.id then output; 
+  keep address_id id casd_date offer_sale_date; 
+  rename address_id=address_id_ref;
+run; 
+
+/** Fill in missing ID numbers to match original TOPA database **/
+data Topa_id_x_address; 
+  merge 
+    Topa_id_x_address_1
+    Prescat.Topa_database (keep=id all_street_addresses address_for_mapping);
+  by id;
+run; 
```

Comment:
I added this data step to fill in the missing IDs from the original TOPA database. By doing a merge by ID, any missing IDs in the Topa_id_x_address_1 will be added to the output data set. Also, I added some vars from the database that I though would be useful in examining the results.